openpathsampling/snapshot.py

```diff
@@ -117,306 +91,73 @@ def copy(self):
+    def copy(self):
+        return super(FeatureSnapshot, self).copy()
 
-# =============================================================================
-# SIMULATION SNAPSHOT (COMPLETE FRAME WITH COORDINATES AND VELOCITIES)
-# =============================================================================
 
-@lazy_loading_attributes('configuration', 'momentum', '_reversed')
-class Snapshot(AbstractSnapshot):
+def snapshot_factory(name, features, description=None):
     """
-    Simulation snapshot. Contains references to a configuration and momentum
+    Helper to create a new Snapshot class
+    
+    Parameters
+    ----------
+    name : str
+        name of the Snapshot class
+    feats : 
     """
 
-    __features__ = [
-        features.configuration,
-        features.momentum
-    ]
-
-
-    def __init__(self, coordinates=None, velocities=None, box_vectors=None,
-                 potential_energy=None, kinetic_energy=None, topology=None,
-                 configuration=None, momentum=None, is_reversed=False,
-                 reversed_copy=None):
-        """
-        Create a simulation snapshot. Initialization happens primarily in
-        one of two ways:
-            1. Specify `Configuration` and `Momentum` objects
-            2. Specify the things which make up `Configuration` and
-               `Momentum` objects, i.e., coordinates, velocities, box
-               vectors, etc.
-        If you want to obtain a snapshot from a currently-running MD engine,
-        use that engine's .current_snapshot property.
-
-        Parameters
-        ----------
-        coordinates : simtk.unit.Quantity wrapping Nx3 np array of dimension length
-            atomic coordinates (default: None)
-        velocities : simtk.unit.Quantity wrapping Nx3 np array of dimension length
-            atomic velocities (default: None)
-        box_vectors : periodic box vectors (default: None)
-            the periodic box vectors at current timestep (defautl: None)
-        potential_energy : simtk.unit.Quantity of units energy/mole
-            potential energy at current timestep (default: None)
-        kinetic_energy : simtk.unit.Quantity of units energy/mole
-            kinetic energy at current timestep (default: None)
-            
-        Attributes
-        ----------
-        coordinates : simtk.unit.Quantity wrapping Nx3 np array of dimension length
-            atomic coordinates
-        velocities : simtk.unit.Quantity wrapping Nx3 np array of dimension length
-            atomic velocities
-        box_vectors : periodic box vectors
-            the periodic box vectors 
-        potential_energy : simtk.unit.Quantity of units energy/mole
-            potential energy
-        kinetic_energy : simtk.unit.Quantity of units energy/mole
-            kinetic energy
-        """
-
-        super(Snapshot, self).__init__(is_reversed, reversed_copy, topology)
+    cls = type(name, (FeatureSnapshot, ), {})
```

Comment:
Thought: would it be ridiculous to add a `superclass` parameter to the factory function? This would default to `FeatureSnapshot`, but would allow simple overrides so that, e.g., I could use a `dynq.Snapshot` (which will be a subclass of `paths.FeatureSnapshot`) as the superclass in `dynamiq_engine`. (There might be extra methods I've implemented for `dynq.Snapshot`, or I might just want the ability to put some specific documentation relevant to the features allowed by `dynamiq_engine`.)